engine/src/flutter/lib/ui/platform_dispatcher.dart

```diff
@@ -1506,6 +1506,88 @@ class PlatformDispatcher {
   // configurationId does not match any configuration.
   @Native<Double Function(Double, Int)>(symbol: 'PlatformConfigurationNativeApi::GetScaledFontSize')
   external static double _getScaledFontSize(double unscaledFontSize, int configurationId);
+
+  /// Color palette preferred by the operating system UI widgets.
+  ///
+  /// On platforms that do not support a UI color palette this value is null.
+  ///
+  /// As of the current release, system colors are supported on web only.
+  ///
+  /// This is typically used in conjunction with
+  /// [AccessibilityFeatures.highContrast]. In particular, on Windows, when
+  /// a user enables high-contrast mode, they may also pick specific colors that
+  /// should be used by application user interfaces.
+  ///
+  /// See also:
+  ///
+  ///   * https://drafts.csswg.org/css-color/#css-system-colors
+  ///   * https://developer.mozilla.org/en-US/docs/Web/CSS/system-color
+  ///   * https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors
+  Map<String, SystemColor>? get systemColors => null;
+
+  /// Standard system color names, as defined by W3C CSS specification.
+  ///
+  /// System color names in Flutter are case-sensitive. This is to make sure
+  /// color names can be easily used as [Map] keys. This is in contrast to CSS,
+  /// where system color names are not case-sensitive. That is, specifying
+  /// `background-color: aCcEnTcOlOr` is equivalent to specifying
+  /// `background-color: AccentColor`. In Flutter,
+  /// `systemColorNames['aCcEnTcOlOr']` will fail to find the respective color.
+  /// Always use the exact name string defined in this list.
+  ///
+  /// See also:
+  ///
+  ///   * https://drafts.csswg.org/css-color/#css-system-colors
+  static const List<String> systemColorNames = <String>[
```

Comment:
a set maybe more concise

Reply:
also is the name here a union between the possible choice between web and windows?

Reply:
The names are W3C standard names, so they are cross-platform. I think it makes sense for us to just adopt the W3C naming, because that will let us cover all platforms, and we won't need to invent our own standard.

Reply:
nvm, I misread some of the doc, I thought we may have this feature in native windows as well

Reply:
We might want to add it for Windows. I just never added it.

Reply:
wait does native windows also have this feature? are you talking about this one? https://support.microsoft.com/en-us/windows/personalize-your-colors-in-windows-3290d30f-d064-5cfe-6470-2fe9c6533e37

Is it that these setting will somehow translate to <system-color> when user user the browser?

Reply:
I can only confirm that the "contrast themes" settings affect system color settings. Changing "accent color" doesn't seem to do anything.